File: Stacks/Stacks/Models/Book.swift
import Foundation

struct Book: Codable, Identifiable {
    let id: Int
    let isbn: String?
    let title: String
    let author: String
    let coverUrl: String?
    let description: String?
    let publishedYear: Int?
    let status: BookStatus?
    let addedAt: Date?
    
    enum CodingKeys: String, CodingKey {
        case id
        case isbn
        case title
        case author
        case coverUrl
        case description
        case publishedYear
        case status
        case addedAt
    }
}

enum BookStatus: String, Codable {
    case wantToRead = "want_to_read"
    case reading = "reading"
    case read = "read"
}

struct BookDetail: Codable {
    let id: Int
    let isbn: String?
    let title: String
    let author: String
    let coverUrl: String?
    let description: String?
    let publishedYear: Int?
    let status: BookStatus?
    let reviews: [Review]
}

File: backend/src/entities/Book.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  OneToMany,
} from 'typeorm';
import { UserBook } from './UserBook';
import { Review } from './Review';

@Entity('books')
export class Book {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ nullable: true, unique: true })
  isbn?: string;

  @Column()
  title!: string;

  @Column()
  author!: string;

  @Column({ name: 'cover_url', nullable: true })
  coverUrl?: string;

  @Column({ type: 'text', nullable: true })
  description?: string;

  @Column({ name: 'published_year', nullable: true })
  publishedYear?: number;

  @OneToMany(() => UserBook, (userBook) => userBook.book)
  userBooks!: UserBook[];

  @OneToMany(() => Review, (review) => review.book)
  reviews!: Review[];
}

